client/src/Views/Landing/GmailDialog.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Displays prompt to authorise MailSender to access their gmail
+ */
+
+import React, { useEffect, useState } from "react";
+import {
+  DialogTitle,
+  Dialog,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  withStyles,
+  Button,
+} from "@material-ui/core";
+
+import googleSignInImg from "Assets/btnGoogleSignIn.png";
+import { getAuthUrl } from "Utils/api";
+
+const styles = () => ({
+  btn: {
+    display: "block",
+    marginLeft: "auto",
+    marginRight: "auto",
+  },
+});
+
+const GmailDialog = props => {
+  const { onClose, open, classes, endRoute } = props;
+
+  const [gmailAuthUrl, setGmailAuthUrl] = useState("");
+
+  useEffect(() => {
+    async function getData() {
+      const authUrl = await getAuthUrl(encodeURIComponent(endRoute));
+      if (authUrl) {
+        setGmailAuthUrl(authUrl);
+      } else {
+        onclose();
+      }
+    }
+    getData();
+  }, [endRoute, onClose]);
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  return (
+    <Dialog onClose={handleClose} open={open}>
```

Comment:
you could just use the onClose function, no need for the handleClose function

Reply:
Ofcourse!